### PR TITLE
Added unit tests for v1 <---> v1alpha1 api conversions

### PR DESCRIPTION
**Pull request by michael81877**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?

Unit tests!

I also corrected the jwt import to use the golang version instead of the other vulnerable one.

### CODE REVIEW

File: api/v1alpha1/nificluster_conversion.go [hunk 1]
```diff
@@ -128,7 +128,7 @@ func convertNifiClusterManagedAdminUsers(src []ManagedUser, dst *v1.NifiCluster)
 func convertNifiClusterManagedReaderUsers(src []ManagedUser, dst *v1.NifiCluster) {
 	dst.Spec.ManagedReaderUsers = []v1.ManagedUser{}
 	for _, user := range src {
-		dst.Spec.ManagedReaderUsers = append(dst.Spec.ManagedAdminUsers, v1.ManagedUser{
```
Comment (michael81877) [inline api/v1alpha1/nificluster_conversion.go:131]: This and the below edits were bugs caught by the added unit tests 🙂

File: api/v1alpha1/common_types.go [hunk 2]
```diff
@@ -49,9 +49,9 @@ type InitClusterNode bool
 type PKIBackend string
 
 // ClientConfigType represents an interface implementing the ClientConfigManager
-type ClientConfigType string
 
 // ClusterType represents an interface implementing the  ClientConfigManager
+type ClientConfigType string
```
Comment (erdrix) [inline api/v1alpha1/common_types.go:54]: A reason for this change ?

**Review by erdrix — APPROVED** [review]
Huge thanks for this PR, not the most fun to do, but of the greatest use !

File: api/v1alpha1/common_types.go (hunk 2 above)
Comment (michael81877) [reply api/v1alpha1/common_types.go]: Purely accidental. I've reverted this. My apologies

**Pull request merged**